src/supporting-data/taxonomy/adapters/taxonomyConverter.ts

```diff
@@ -16,19 +19,27 @@ type Strain = {
   name: string;
 };
 
-type TaxonomyLight = {
-  taxonId: number;
-  scientificName: string;
-  synonyms?: string[];
-  commonName?: string;
-};
+export type Lineage = Array<SetRequired<TaxonomyDatum, 'hidden' | 'rank'>>;
 
-type TaxonomyLightWithHiddenFlag = TaxonomyLight & {
-  hidden: boolean;
+// Temporary function to check lineage type while backend unifies its model
+export const isLineage = (lineage: Lineage | string[]): lineage is Lineage => {
+  if ((lineage as Lineage).some((item) => item.taxonId)) {
+    return true;
+  }
+  return false;
 };
 
-type TaxonomyLightWithMnemonic = TaxonomyLight & {
-  mnemonic: string;
+export type TaxonomyDatum = {
+  taxonId: number;
+  scientificName?: string;
+  synonyms?: string[];
+  commonName?: string;
+  mnemonic?: string;
+  evidences?: Evidence[];
+  // Note: eventually should all be an array of TaxonomyData
+  lineage?: Lineage | string[];
```

Comment:
Probably should mark it as `@deprecated`, so that in the places where we use it as an array of string, the editor will remind us it is deprecated.
Something like this would cover it:
```ts
type TaxonomyBase = {
  taxonId: number;
  scientificName?: string;
  synonyms?: string[];
  commonName?: string;
  mnemonic?: string;
  evidences?: Evidence[];
  lineage?: Lineage;
  hidden?: boolean;
  rank?: Rank;
};

export type TaxonomyDatum =
  | TaxonomyBase
  | (TaxonomyBase & {
      /**
       * @deprecated "lineage" should be a `Lineage`
       */
      lineage?: string[];
    });
```